### DISCUSSION TITLE
Plugin Activation Errors

### DISCUSSION THREAD

**Discussion by theotillotson** [impress-org/givewp#6622] [Developer Q&A]
Plugin Activation Errors for PHP 8.0:

( ! ) Warning: Undefined array key "gateways" in /wp-content/plugins/give/src/PaymentGateways/Gateways/PayPalStandard/Migrations/SetPayPalStandardGatewayId.php on line 27
--

1 | 0.0002 | 364000 | {main}( ) | .../plugins.php:0
2 | 0.1554 | 14884688 | do_action( $hook_name = 'activate_give/give.php' ) | .../plugins.php:194
3 | 0.1554 | 14885064 | WP_Hook->do_action( $args = [0 => ''] ) | .../plugin.php:517
4 | 0.1554 | 14885064 | WP_Hook->apply_filters( $value = '', $args = [0 => ''] ) | .../class-wp-hook.php:332
5 | 0.1554 | 14886192 | Give->install( '' ) | .../class-wp-hook.php:308
6 | 0.1635 | 16167032 | give_install( $network_wide = ??? ) | .../give.php:396
7 | 0.1635 | 16167032 | give_run_install( ) | .../install.php:49
8 | 0.1750 | 16269304 | do_action( $hook_name = 'give_upgrades' ) | .../install.php:109
9 | 0.1750 | 16269680 | WP_Hook->do_action( $args = [0 => ''] ) | .../plugin.php:517
10 | 0.1750 | 16269680 | WP_Hook->apply_filters( $value = '', $args = [0 => ''] ) | .../class-wp-hook.php:332
11 | 0.1751 | 16270808 | Give\Helpers\Hooks::Give\Helpers\{closure:/wp-content/plugins/give/src/Helpers/Hooks.php:32-44}( '' ) | .../class-wp-hook.php:308
12 | 0.1756 | 16319040 | call_user_func_array:{/wp-content/plugins/give/src/Helpers/Hooks.php:43}( $callback = [0 => class Give\Framework\Migrations\MigrationsRunner { private $completedMigrations = [...]; private $migrationRegister = class Give\Framework\Migrations\MigrationsRegister { ... }; private $migrationLogFactory = class Give\MigrationLog\MigrationLogFactory { ... }; private $migrationLogRepository = class Give\MigrationLog\MigrationLogRepository { ... } }, 1 => 'run'], $args = [0 => ''] ) | .../Hooks.php:43
13 | 0.1756 | 16319112 | Give\Framework\Migrations\MigrationsRunner->run( '' ) | .../Hooks.php:43
14 | 0.1759 | 16301328 | Give\PaymentGateways\Gateways\PayPalStandard\Migrations\SetPayPalStandardGatewayId->run( ) | .../MigrationsRunner.php:113

( ! ) Warning: Undefined array key "gateways" in /wp-content/plugins/give/src/PaymentGateways/Gateways/PayPalStandard/Migrations/SetPayPalStandardGatewayId.php on line 27
Call Stack
#	Time	Memory	Function	Location
1	0.0002	364000	{main}( )	.../plugins.php:0
2	0.1554	14884688	do_action( $hook_name = 'activate_give/give.php' )	.../plugins.php:194
3	0.1554	14885064	WP_Hook->do_action( $args = [0 => ''] )	.../plugin.php:517
4	0.1554	14885064	WP_Hook->apply_filters( $value = '', $args = [0 => ''] )	.../class-wp-hook.php:332
5	0.1554	14886192	Give->install( '' )	.../class-wp-hook.php:308
6	0.1635	16167032	give_install( $network_wide = ??? )	.../give.php:396
7	0.1635	16167032	give_run_install( )	.../install.php:49
8	0.1750	16269304	do_action( $hook_name = 'give_upgrades' )	.../install.php:109
9	0.1750	16269680	WP_Hook->do_action( $args = [0 => ''] )	.../plugin.php:517
10	0.1750	16269680	WP_Hook->apply_filters( $value = '', $args = [0 => ''] )	.../class-wp-hook.php:332
11	0.1751	16270808	Give\Helpers\Hooks::Give\Helpers\{closure:/wp-content/plugins/give/src/Helpers/Hooks.php:32-44}( '' )	.../class-wp-hook.php:308
12	0.1756	16319040	[call_user_func_array:{/wp-content/plugins/give/src/Helpers/Hooks.php:43}](http://www.php.net/function.call-user-func-array:%7B/wp-content/plugins/give/src/Helpers/Hooks.php:43%7D)( $callback = [0 => class Give\Framework\Migrations\MigrationsRunner { private $completedMigrations = [...]; private $migrationRegister = class Give\Framework\Migrations\MigrationsRegister { ... }; private $migrationLogFactory = class Give\MigrationLog\MigrationLogFactory { ... }; private $migrationLogRepository = class Give\MigrationLog\MigrationLogRepository { ... } }, 1 => 'run'], $args = [0 => ''] )	.../Hooks.php:43
13	0.1756	16319112	Give\Framework\Migrations\MigrationsRunner->run( '' )	.../Hooks.php:43
14	0.1759	16301328	Give\PaymentGateways\Gateways\PayPalStandard\Migrations\SetPayPalStandardGatewayId->run( )	.../MigrationsRunner.php:113

( ! ) Fatal error: Uncaught TypeError: array_key_exists(): Argument #2 ($array) must be of type array, null given in /wp-content/plugins/give/src/PaymentGateways/Gateways/PayPalStandard/Migrations/SetPayPalStandardGatewayId.php on line 30
( ! ) TypeError: array_key_exists(): Argument #2 ($array) must be of type array, null given in /wp-content/plugins/give/src/PaymentGateways/Gateways/PayPalStandard/Migrations/SetPayPalStandardGatewayId.php on line 30
Call Stack
#	Time	Memory	Function	Location
1	0.0002	364000	{main}( )	.../plugins.php:0
2	0.1554	14884688	do_action( $hook_name = 'activate_give/give.php' )	.../plugins.php:194
3	0.1554	14885064	WP_Hook->do_action( $args = [0 => ''] )	.../plugin.php:517
4	0.1554	14885064	WP_Hook->apply_filters( $value = '', $args = [0 => ''] )	.../class-wp-hook.php:332
5	0.1554	14886192	Give->install( '' )	.../class-wp-hook.php:308
6	0.1635	16167032	give_install( $network_wide = ??? )	.../give.php:396
7	0.1635	16167032	give_run_install( )	.../install.php:49
8	0.1750	16269304	do_action( $hook_name = 'give_upgrades' )	.../install.php:109
9	0.1750	16269680	WP_Hook->do_action( $args = [0 => ''] )	.../plugin.php:517
10	0.1750	16269680	WP_Hook->apply_filters( $value = '', $args = [0 => ''] )	.../class-wp-hook.php:332
11	0.1751	16270808	Give\Helpers\Hooks::Give\Helpers\{closure:/wp-content/plugins/give/src/Helpers/Hooks.php:32-44}( '' )	.../class-wp-hook.php:308
12	0.1756	16319040	[call_user_func_array:{/wp-content/plugins/give/src/Helpers/Hooks.php:43}](http://www.php.net/function.call-user-func-array:%7B/wp-content/plugins/give/src/Helpers/Hooks.php:43%7D)( $callback = [0 => class Give\Framework\Migrations\MigrationsRunner { private $completedMigrations = [...]; private $migrationRegister = class Give\Framework\Migrations\MigrationsRegister { ... }; private $migrationLogFactory = class Give\MigrationLog\MigrationLogFactory { ... }; private $migrationLogRepository = class Give\MigrationLog\MigrationLogRepository { ... } }, 1 => 'run'], $args = [0 => ''] )	.../Hooks.php:43
13	0.1756	16319112	Give\Framework\Migrations\MigrationsRunner->run( '' )	.../Hooks.php:43
14	0.1759	16301328	Give\PaymentGateways\Gateways\PayPalStandard\Migrations\SetPayPalStandardGatewayId->run( )	.../MigrationsRunner.php:113
15	0.1761	16302880	[array_key_exists](http://www.php.net/function.array-key-exists)( $key = 'paypal-standard', $array = NULL )	.../SetPayPalStandardGatewayId.php:30

**Comment by JoshuaHungDinh** [comment]
Hey @theotillotson! Seeing as this seems to be a potential code issue. I've started a feedback report to keep track of the error. It can be found at https://feedback.givewp.com/bug-reports/p/php-80-plugin-activation-errors. Feel free to include any additional information you feel might be helpful.  Thanks for bringing this error to our attention and happy holidays!

**Reply by JoshuaHungDinh** [reply]
Hi @theotillotson, I wanted to circle back to you in regards to your activation error.  After consulting with our team we believe the quickest way to a resolution for your specific error will be by contacting our priority support team.  If the error persist's you can reach them at  https://givewp.com/support/.

**Comment by wspf0824** [comment]
这是来自QQ邮箱的假期自动回复邮件。
 
您好，我最近正在休假中，无法亲自回复您的邮件。我将在假期结束后，尽快给您回复。
